src/JNCC.PublicWebsite.Core/Services/CSVExportService.cs

```diff
@@ -0,0 +1,142 @@
+using System.Web;
+using Umbraco.Core.Services;
+using Umbraco.Core.Logging;
+
+namespace JNCC.PublicWebsite.Core.Services
+{
+	public static class CSVExportService
+	{
+
+		/// <summary>
+		/// Parses a list of objects and returns a csv file with all the correct header information and property values. 
+		/// </summary>
+		/// <param name="dList">The list to format</param>
+		/// <param name="FileName">The name of the csv file (a timestamp is added)</param>
+		public static void GenericExport(object[] dList, string FileName, bool SpaceCapitals = false)
+		{
+			string data = GenerateCSVData(dList, SpaceCapitals);
+
+			HttpContext.Current.Response.Clear();
+			HttpContext.Current.Response.ContentType = "text/csv";
+			HttpContext.Current.Response.AddHeader("content-disposition", "attachment; filename=" + FileName + DateTime.UtcNow.ToString("dd-MM-yyyy") + ".csv");
+			HttpContext.Current.Response.Write(data);
+			HttpContext.Current.Response.End();
+
+		}
+
+		/// <summary>
+		/// Generates a CSV data string for the given params
+		/// </summary>
+		/// <param name="dList"></param>
+		/// <param name="SpaceCapitals"></param>
+		/// <returns></returns>
+		public static string GenerateCSVData(object[] dList, bool SpaceCapitals = false)
+		{
+			if (!dList.Any())
+			{ return ""; }
+
+			var input = dList.First();
+
+			Type dType = input.GetType();
+
+			string HeaderText = "";
+
+			//loop through the properties and format the header of the csv file
+			//Any item will do, they all have the same properties.
+			int limit = dType.GetProperties().Count();
+			int count = 0;
+			foreach (var prop in dType.GetProperties())
+			{
+				if (SpaceCapitals)
+				{ HeaderText += AddSpacesToSentence(prop.Name, true); }
+				else
+				{ HeaderText += prop.Name; }
+
+				//If the first letters of a csv is 'ID', excel assumes you're trying to open an SYLK file.
+				//To avoid formatting errors, enclose an ID in quotes (which formats normally)
+				if (HeaderText.ToLower() == "id")
+				{ HeaderText = "\"" + HeaderText + "\""; }
+
+				count++;
+
+				if (count != limit)
+				{ HeaderText += ","; }
+
+			}
+
+			var sbOutput = new StringBuilder();
+
+			sbOutput.Append(HeaderText + "\r\n");
+
+			//Go through each item in list
+			foreach (object item in dList)
+			{
+				Type itmType = item.GetType();
+
+				//Go through each property of each item
+				foreach (PropertyInfo pi in itmType.GetProperties())
+				{
+					object propValue = pi.GetValue(item, null);
+
+					var piType = pi.PropertyType;
+					if (propValue == null)
+					{
+						sbOutput.Append(FormatLine("NULL"));
+					}
+					else
+					{
+						//Format certain values certain ways.
+						if (piType.IsEquivalentTo(typeof(DateTime)))
+						{
+							sbOutput.Append(FormatLine(((DateTime)propValue).ToString("dd.MM.yy")));
+						}
+						else if (piType.IsEquivalentTo(typeof(decimal)))
+						{
+							sbOutput.Append(FormatLine(((decimal)propValue).ToString("N2")));

```

Comment:
Is it right to always assume 2dp?

Reply:
For decimals, this always seems reasonable